tools/autograd/templates/ViewFuncs.cpp

```diff
@@ -0,0 +1,69 @@
+#include "torch/csrc/autograd/generated/ViewFuncs.h"
+
+// ${generated_comment}
+
+using at::Tensor;
+using at::Scalar;
+using at::IntArrayRef;
+using at::TensorList;
+
+namespace torch::autograd {
+
+ChainedViewFunc::ChainedViewFunc(
+  const std::shared_ptr<ViewFunc>& first_,
+  const std::shared_ptr<ViewFunc>& second_)
+  : first(first_),
+    num_first_symints(-1),
+    num_first_tensors(-1),
+    second(second_),
+    num_second_symints(-1),
+    num_second_tensors(-1) {}
+
+std::vector<c10::SymInt> ChainedViewFunc::get_symints() {
+  auto symints = first->get_symints();
+  num_first_symints = symints.size();
+  auto second_symints = second->get_symints();
+  num_second_symints = second_symints.size();
+  symints.reserve(symints.size() + num_second_symints);
+  symints.insert(symints.end(), second_symints.begin(), second_symints.end());
```

Comment:
yep that's right - sizes must be the same but otherwise arbitrary modification

Reply:
Given how we use it for chaining I agree it is the right contract to avoid creating too many Vectors.
Please add a comment on the base class to make it clear!

Reply:
You got it!